lib/Controller/ShareApiController.php

```diff
@@ -0,0 +1,199 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Forms\Controller;
+
+use OCA\Forms\Constants;
+use OCA\Forms\Db\Form;
+use OCA\Forms\Db\FormMapper;
+use OCA\Forms\Db\Share;
+use OCA\Forms\Db\ShareMapper;
+use OCA\Forms\Service\FormsService;
+
+use OCP\AppFramework\OCSController;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\IMapperException;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCS\OCSBadRequestException;
+use OCP\AppFramework\OCS\OCSException;
+use OCP\AppFramework\OCS\OCSForbiddenException;
+use OCP\ILogger;
+use OCP\IRequest;
+use OCP\IUser;
+use OCP\IUserSession;
+use OCP\Security\ISecureRandom;
+use OCP\Share\IShare;
+
+class ShareApiController extends OCSController {
+	protected $appName;
+
+	/** @var FormMapper */
+	private $formMapper;
+
+	/** @var ShareMapper */
+	private $shareMapper;
+
+	/** @var FormsService */
+	private $formsService;
+
+	/** @var ILogger */
+	private $logger;
+	
+	/** @var IUser */
+	private $currentUser;
+
+	/** @var ISecureRandom */
+	private $secureRandom;
+
+	public function __construct(string $appName,
+								FormMapper $formMapper,
+								ShareMapper $shareMapper,
+								FormsService $formsService,
+								ILogger $logger,
+								IRequest $request,
+								IUserSession $userSession,
+								ISecureRandom $secureRandom) {
+		parent::__construct($appName, $request);
+		$this->appName = $appName;
+		$this->formMapper = $formMapper;
+		$this->shareMapper = $shareMapper;
+		$this->formsService = $formsService;
+		$this->logger = $logger;
+		$this->secureRandom = $secureRandom;
+
+		$this->currentUser = $userSession->getUser();
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Add a new share
+	 *
+	 * @param int $formId The form to share
+	 * @param int $shareType Nextcloud-ShareType
+	 * @param string $shareWith ID of user/group/... to share with. For Empty shareWith and shareType Link, this will be set as RandomID.
+	 * @return DataResponse
+	 * @throws OCSBadRequestException
+	 * @throws OCSForbiddenException
+	 */
+	public function newShare(int $formId, int $shareType, string $shareWith = ''): DataResponse {
+		$this->logger->debug('Adding new share: formId: {formId}, shareType: {shareType}, shareWith: {shareWith}', [
+			'formId' => $formId,
+			'shareType' => $shareType,
+			'shareWith' => $shareWith,
+		]);
+
+		// Only accept usable shareTypes
+		if (array_search($shareType, Constants::SHARE_TYPES_USED) === false) {
+			$this->logger->debug('Invalid shareType');
+			throw new OCSBadRequestException('Invalid shareType');
+		}
+
+		try {
+			$form = $this->formMapper->findById($formId);
+		} catch (IMapperException $e) {
+			$this->logger->debug('Could not find form');
```

Comment:
```suggestion
			$this->logger->debug('Could not find form', ['exception' => $e]);
```
Most of the time if you have an exception for the context it’s good to pass it to the logger. (only putting that comment here but other try/catch are concerned)

Reply:
👍